Automattic/jetpack
# Eventbrite Block: Update Button

Fixes #15710

#### Changes proposed in this Pull Request:

* Replaces the Eventbrite block's `ModalButtonPreview` component with the `Button` inner block.
* Update the Button block to support fill and outline styles, and inline formats (required by Eventbrite and others).

| Before | After |
| - | - |
| <img width="884" alt="Screenshot 2020-05-25 at 15 42 54" src="https://user-images.githubusercontent.com/2070010/82823111-e14a8480-9e9e-11ea-8193-05dce040d6e2.png"> | <img width="849" alt="Screenshot 2020-05-25 at 15 42 25" src="https://user-images.githubusercontent.com/2070010/82823121-ea3b5600-9e9e-11ea-91ff-0a25dcc72f4a.png"> <img width="863" alt="Screenshot 2020-05-25 at 15 54 37" src="https://user-images.githubusercontent.com/2070010/82823789-1c00ec80-9ea0-11ea-971e-5ddd002ce2a4.png"> |

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Before checking out this PR:
  * Insert two Eventbrite block, and embed an event for both of them.
  * Change the style of one of the block to "Button & Modal", while leaving the other as "In-page Embed".
  * Change the "Button & Modal" block's settings as much as possible: colors, border radius, text, style.
  * Save.
* Checkout this PR, and reload the post.
* Look at the browser console. Make sure that only one of the two blocks (the "Button & Modal" one) has migrated.
* Make sure the block and all the customizations have migrated correctly.
* Save and preview.
* Make sure the button works as expected: clicking it will open an Eventbrite modal (over HTTPS only, otherwise it will simply opens an Eventbrite page in a new tab).

#### Proposed changelog entry for your changes:

* Eventbrite block: Use new Button inner block.
* Button block: Add support for button styles and inline formats.
